Getting Started grammar review (#16)
hanami/guides: content/introduction/getting-started.md

## Hanami's Architecture

Hanami's architecture revolves around your project containing many `apps`.
These all live together in the same codebase, and exist in the same Ruby process.

They live under `apps/`.

By default, we have a `web` app, which can be thought of as the standard, user-facing web interface.
This is the most popular, so you'll probably want to keep it in your future Hanami projects.
However, there's nothing unique about this app, it's just so common that Hanami generates it for us.

> Later (in a real project), we would add other apps, such as an `admin` panel, a JSON `API`, or an analytics `dashboard`.

Reviewer: ```suggestion
Later (in a real project), we would add other apps, such as an `admin` panel, a JSON `api`, or an analytics `dashboard`.
```
I agree API should be capitalized, but this context is the name of the app, which _has_(?) to be lowercase.

> We could also break our `web` app into smaller.

Reviewer: ```suggestion
We could also break our `web` app into smaller parts, extracting isolated pieces of functionality.
```
Just noticed this sentence was incomplete :)

Hanami fully supports that, too!

Different `apps` represent __delivery mechanisms__.
That means they're different ways of interacting with the core of your project, or the "business logic".

> Hanami doesn't want us to [repeat ourselves](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) so that "business logic" is shared.

Reviewer: ```suggestion
Hanami doesn't want us to [repeat ourselves](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself), so that "business logic" is shared.
```
I think the comma helps here a lot, and changes the meaing of the sentence. Perhaps we could find another way to re-word this so that it's more clear and we can avoid a single comma changing the meaning. What do you think?

```

This is an empty action that doesn't do anything special.

> Each action in Hanami is defined by [its class](https://en.wikipedia.org/wiki/Single_responsibility_principle), which makes it simple to test.

Reviewer: ```suggestion
Each action in Hanami is defined by [a single class](https://en.wikipedia.org/wiki/Single_responsibility_principle), which makes it simple to test.
```
I think this suggestion gets to the point we're trying to make here. (That it's different than Rails where you can have 7+ actions in a class)

- Creating a view at `apps/web/views/books/index.rb` (and a spec for it),
- Creating a template at `apps/web/templates/books/index.html.erb`.

> _(If you're confused by 'action' vs. 'controller,' here's a tip: Hanami only has `action` classes, so a controller is just a module to group several related actions together.)_

Reviewer: ```suggestion
_(If you're confused by 'action' vs. 'controller': Hanami only has `action` classes, so a controller is just a module to group several related actions together.)_
```

* an **entity** is a domain object (a `Book`) that is uniquely identified by its identity,
* a **repository** is what we use to persist, retrieve, and delete data for an entity, in the persistence layer.

> Entities are unaware of the database.

Reviewer: Hm, I think this is a pretty big distinction so I like the adverb for emphasis. Maybe `entirely` instead of `totally`, what do you think?

### Working With Entities

> An entity is something really close to an everyday Ruby object.

Reviewer: ```suggestion
An entity is something really close to a plain Ruby object.
```
I think the word `plain` is better here, since "Plain old Ruby object" is [a common term](https://codesthq.com/hi-im-poro/), and often abbreviated as "PORO" (though I don't think we shoud introduce that acronym here).

And it comes from Java, where it's also [a common term](https://en.wikipedia.org/wiki/Plain_old_Java_object)

Author: Why do you want to avoid using PORO? That would have clarified it for me a bit.

### Using Repositories

Now we are ready to play around with our repository.

> We can use Hanami's `console` command to launch irb with our application pre-loaded, so we can use our objects:

Reviewer: ```suggestion
We can use Hanami's `console` command to launch `irb` with our application pre-loaded, so we can use our objects:
```
